frontend/src/Context/PublisherProvider/usePublisher/usePublisher.tsx

```diff
@@ -162,31 +172,35 @@ const usePublisher = (): PublisherContextType => {
     }
   };
 
-  const addPublisherListeners = (publisher: Publisher) => {
+  const addPublisherListeners = useCallback((publisher: Publisher) => {
     publisher.on('destroyed', handleDestroyed);
     publisher.on('streamCreated', handleStreamCreated);
     publisher.on('streamDestroyed', handleStreamDestroyed);
     publisher.on('accessDenied', handleAccessDenied);
     publisher.on('videoElementCreated', handleVideoElementCreated);
     publisher.on('muteForced', handleMuteForced);
     publisher.on('accessAllowed', handleAccessAllowed);
-  };
+  }, []);
 
   /**
    * Method to create local camera publisher.
+   * @param {PublisherProperties} options - the publisher options to initialize the local publisher with
    */
-  const initializeLocalPublisher = () => {
-    try {
-      const publisher = initPublisher(undefined, publisherOptions);
-      // Add listeners synchronously as some events could be fired before callback is invoked
-      addPublisherListeners(publisher);
-      publisherRef.current = publisher;
-    } catch (error) {
-      if (error instanceof Error) {
-        console.warn(error.stack);
+  const initializeLocalPublisher = useCallback(
+    (options?: PublisherProperties) => {
+      try {
+        const publisher = initPublisher(undefined, options);
```

Comment:
if options is undefined do we want to be doing this?

Reply:
changed the options to be required and not optional so going to resolve this one and I think another comment around this 🙏